### ISSUE TITLE
Can we have Spark 3 Enceladus that talks both to Hadoop 2.7.3 and 3?

### ISSUE THREAD

**Issue by Zejnilovic** [AbsaOSS/enceladus#1746]
## Question
If we move to Spark 3 we still have our corp's infrastructure to be compatible with. This means at least being able to work with Hadoop 2.7.3 (on prem) and 3 (on EMR). Is this binary-ly feasible?

How harder/easier might it be to support both spark 2.4 and 3?

**Comment by AdrianOlosutean** [comment]
After having a look [here](https://spark.apache.org/releases/spark-release-3-0-0.html), I noticed that only 2.6 support was removed.  However, this known issue could be particularly important to know:

- With AWS SDK upgrade to 1.11.655, we strongly encourage the users that use S3N file system (open-source NativeS3FileSystem that is based on jets3t library) on Hadoop 2.7.3 to upgrade to use AWS Signature V4 and set the bucket endpoint or migrate to S3A (“s3a://” prefix) - jets3t library uses AWS v2 by default and s3.amazonaws.com as an endpoint. Otherwise, the 403 Forbidden error may be thrown in the following cases:
If a user accesses an S3 path that contains “+” characters and uses the legacy S3N file system, e.g. s3n://bucket/path/+file.
If a user has configured AWS V2 signature to sign requests to S3 with S3N file system.
Note that if you use S3AFileSystem, e.g. (“s3a://bucket/path”) to access S3 in S3Select or SQS connectors, then everything will work as expected. (SPARK-30968)
